### ISSUE TITLE
Set up permanent SWIB chat

### ISSUE THREAD

**Issue by acka47** [swibcon/swib-orga#12]
Annif workshop facilitators have asked when this chat is available so that they can use it for communication with their participants

Der Chat sollte 2023 noch an den Start gebracht werden.

**Comment by sroertgen** [comment]
Who is willing to host the service? Or should it be a hosted service?

**Comment by acka47** [comment]
I think we should go for a hosted Mattermost or Matrix server at [collochat.de](https://collochat.de/). We won't get this running quickly enough as a self-hosted version.

**Comment by acka47** [comment]
Just discussed this with @sroertgen and had a strong tendency towards mattermost because:

- This is what we have been using in the SWIB context already.
- It is DSGVO conform and easily – trhough BBB integration – allows video conferences with more than 2 participants.
- We might make it Matrix-conformant later with https://mattermost.com/marketplace/mattermost-matrix-connector/

Next step: Clear financing.

**Comment by acka47** [comment]
We have received a quote in November but also thought about maybe using metadaten.community for SWIB as it primarily offers a forum for in depth professional discussions but also a chat to be used for rather organizational conference communication.

However, @Phu2 and me have discussed the option and rejected the idea for these reasons:

- As metadaten.community is a German-speaking forum it would be quite confusing for SWIB participants if the conference discussion happened in this context.
- We'd have to translate  whole privacy and terms of service to English.
- It might be difficult to keep the German-speaking focus in the forum with lots of non-German-speaking people registered through SWIB.

Currently, we tend towards starting a Mattermost chat and to evaluate its usage after one year and check whether the benefits justify the costs. Another option to assess would be setting up a dedicated discourse forum for SWIB which would be no great burden wrt technical maintenance.

**Comment by acka47** [comment]
Just a note: A general problem with a permanent SWIB communication platform – whether Mattermost or Discourse – is that we will have problems linking registered SWIB participants with their profile. In the past this was easy because we set up a new Mattermost for each SWIB and automatically registered a corresponding Mattermost user with every registrant in ConfTool.

**Comment by acka47** [comment]
Plan: 
- We will set up a Mattermost chat together with ConfTool around the time of the CfP.
- We will have to check whether it makes sense and is technically feasible to set up a Mattermost account not upon registration for the conferen but upon user registration in ConfTool.

**Comment by acka47** [comment]
This time, we should have the chat running unter `https://chat.swib.org`.

**Comment by acka47** [comment]
In the last weeks we have been evaluating [Discourse](https://discourse.org/) as a platform for SWIB discussions and finally decided to use it instead of Mattermost. Reasons:

- We will also use Discourse for conference and workshop registration so that participants will only have to interact with two systems (website & discourse) instead of three (ConfTool, website, Mattemrost).
- With (workshop) registration and communication being coupled in one application, we won't get the synching problems between the registration system (ConfTool in the past) and the chat/discussion system (until now Mattermost) which in the past could only be resolved by regular manual intervention.
- We are already hosting metadaten.community, so that adding another Discourse instance isn't that much of an effort.
- The costs will e significantly lower and we are able to control the setup without much communicational overhead (compared to communicating changes with an external party).

Furthermore, we evaluated replacing current swib.org setup with discourse as well. As this would be too much of a break and there are a lot of open questions, we will now longer evaluate this option and stay with the current approach for building swib.org and the programme. As we will carry on using  ConfTool  for registration and programme committee work we will just use the established export and build processes.

Of course, there are still some open questions re. Discourse (e.g. #16) which we plan to resolve the coming weeks. I have informed Collocall that we will not commission a Mattermost chat this year. Closing.
